### PR TITLE
2310.1 Point Release

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
## What does this PR do?

Merge `point-release/23101` to `main`

## How was this PR tested?

- no changes have been made, nor were there any merge conflicts
- all testing  was done on the `point-release/23101` branch

### CODE REVIEW

**Review by AMZN-daimini — APPROVED** [review]

File: scripts/o3de/ExportScripts/export_source_built_project.py [hunk 1]
```diff
@@ -231,27 +390,77 @@ def parse_args(o3de_context: exp.O3DEScriptExportContext):
...
+                                                                           key=exp.SETTINGS_OPTION_BUILD_SERVER_LAUNCHER.key,
+                                                                           enable_attribute='server_launcher',
+                                                                           disable_attribute='no_server_launcher')
+
+    option_build_headless_server_launcher = export_config.get_parsed_boolean_option(parsed_args=args,
+                                                                                    key=exp.SETTINGS_OPTION_BUILD_HEADLESS_SERVER_LAUNCHER.key,
+                                                                                    enable_attribute='headless_server_launcher',
+                                                                                    disable_attribute='no_headless_server_launcher')
+
+    option_build_unified_launcher = export_config.get_parsed_boolean_option(parsed_args=args,
```
Comment (spham-amzn) [inline scripts/o3de/ExportScripts/export_source_built_project.py:424]: Headless server is not supported in 2310 yet. 
```suggestion
```

File: scripts/o3de/ExportScripts/export_source_built_project.py [hunk 2]
```diff
@@ -30,6 +36,7 @@ def export_standalone_project(ctx: exp.O3DEScriptExportContext,
                               should_build_all_assets: bool = True,
                               should_build_game_launcher: bool = True,
                               should_build_server_launcher: bool = True,
+                              should_build_headless_server_launcher: bool = True,
```
Comment (spham-amzn) [inline scripts/o3de/ExportScripts/export_source_built_project.py:39]: Headless server is not supported in 2310 yet.
```suggestion
```

File: scripts/o3de/ExportScripts/export_source_built_project.py [hunk 3]
```diff
@@ -103,6 +125,8 @@ def export_standalone_project(ctx: exp.O3DEScriptExportContext,
         launcher_type |= exp.LauncherType.SERVER
     if should_build_unified_launcher:
         launcher_type |= exp.LauncherType.UNIFIED
+    if should_build_headless_server_launcher:
+        launcher_type |= exp.LauncherType.HEADLESS_SERVER
```
Comment (spham-amzn) [inline scripts/o3de/ExportScripts/export_source_built_project.py:129]: Headless server is not supported in 2310 yet.
```suggestion
```

File: scripts/o3de/ExportScripts/export_source_built_project.py [hunk 4]
```diff
@@ -231,27 +390,77 @@ def parse_args(o3de_context: exp.O3DEScriptExportContext):
...
+
+    option_build_server_launcher = export_config.get_parsed_boolean_option(parsed_args=args,
+                                                                           key=exp.SETTINGS_OPTION_BUILD_SERVER_LAUNCHER.key,
+                                                                           enable_attribute='server_launcher',
+                                                                           disable_attribute='no_server_launcher')
+
+    option_build_headless_server_launcher = export_config.get_parsed_boolean_option(parsed_args=args,
+                                                                                    key=exp.SETTINGS_OPTION_BUILD_HEADLESS_SERVER_LAUNCHER.key,
+                                                                                    enable_attribute='headless_server_launcher',
+                                                                                    disable_attribute='no_headless_server_launcher')
```
Comment (spham-amzn) [inline scripts/o3de/ExportScripts/export_source_built_project.py:422]: Headless server is not supported in 2310 yet.
```suggestion
```

File: scripts/o3de/o3de/export_project.py [hunk 5]
```diff
@@ -186,6 +191,7 @@ class LauncherType(IntEnum):
     GAME = 1
     SERVER = 2
     UNIFIED = 4
+    HEADLESS_SERVER = 8
```
Comment (spham-amzn) [inline scripts/o3de/o3de/export_project.py:194]: Headless Server not support in 2310 
```suggestion
```

File: scripts/o3de/o3de/export_project.py [hunk 6]
```diff
@@ -535,6 +717,7 @@ def build_game_targets(ctx: O3DEScriptExportContext,
     should_build_game_launcher = (launcher_types & LauncherType.GAME) == LauncherType.GAME
     should_build_server_launcher = (launcher_types & LauncherType.SERVER) == LauncherType.SERVER
     should_build_unified_launcher = (launcher_types & LauncherType.UNIFIED) == LauncherType.UNIFIED
+    should_build_headless_server_launcher = (launcher_types & LauncherType.SERVER) == LauncherType.HEADLESS_SERVER
```
Comment (spham-amzn) [inline scripts/o3de/o3de/export_project.py:720]: Headless Server not supported in 2310
```suggestion
```

File: scripts/o3de/o3de/export_project.py [hunk 7]
```diff
@@ -577,6 +760,8 @@ def build_game_targets(ctx: O3DEScriptExportContext,
         mono_build_args.append(f"{ctx.project_name}.GameLauncher")
     if should_build_unified_launcher:
         mono_build_args.append(f"{ctx.project_name}.UnifiedLauncher")
+    if should_build_headless_server_launcher:
+        mono_build_args.append(f"{ctx.project_name}.HeadlessServerLauncher")
```
Comment (spham-amzn) [inline scripts/o3de/o3de/export_project.py:764]: Headless Server not supported in 2310
```suggestion
```

File: scripts/o3de/o3de/export_project.py [hunk 8]
```diff
@@ -777,3 +978,169 @@ def validate_project_artifact_paths(project_path: pathlib.Path,
...
+
+SETTINGS_OPTION_BUILD_SERVER_LAUNCHER = register_setting(key='option.build.server.launcher',
+                                                         description='The option to build an exported server launcher.',
+                                                         is_boolean=True,
+                                                         default='True')
+
+SETTINGS_OPTION_BUILD_HEADLESS_SERVER_LAUNCHER = register_setting(key='option.build.headless.server.launcher',
+                                                                  description='The option to build an exported headless server launcher.',
+                                                                  is_boolean=True,
+                                                                  default='False')
```
Comment (spham-amzn) [inline scripts/o3de/o3de/export_project.py:1121]: Headless Server not supported in 2310
```suggestion
```

File: scripts/o3de/ExportScripts/export_source_built_project.py [hunk 9]
```diff
@@ -184,40 +231,152 @@ def parse_args(o3de_context: exp.O3DEScriptExportContext):
...
+                                           disable_override_arg=['--nounified', '--no-unified-launcher'],
+                                           disable_override_desc="Disable the building and inclusion of the Unified Launcher if not needed.")
+
+        export_config.add_boolean_argument(parser=parser,
+                                           key=exp.SETTINGS_OPTION_ENGINE_CENTRIC.key,
+                                           enable_override_arg=['-ec', '--engine-centric'],
+                                           enable_override_desc="Enable the engine-centric work flow to export the project.",
+                                           disable_override_arg=['-pc', '--project-centric'],
+                                           disable_override_desc="Enable the project-centric work flow to export the project.")
+
```
Comment (spham-amzn) [inline scripts/o3de/ExportScripts/export_source_built_project.py:379]: Missing `platforms` argument
```suggestion
        parser.add_argument('-pl', '--platform', type=str, default=exp.get_default_asset_platform(), choices=['pc', 'linux', 'mac'])

```

**Comment by AMZN-alexpete** [comment]
Thanks @spham-amzn! I'll pull in the changes from https://github.com/o3de/o3de/pull/17188 after it is merged to resolve mentioned issues.

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**